runtime/src/main/java/org/corfudb/runtime/view/stream/AbstractQueuedStreamView.java

```diff
@@ -105,20 +113,84 @@ protected ILogData getNextEntry(QueuedStreamContext context,
         }
 
         // Otherwise we remove entries one at a time from the read queue.
-        if (getFrom.size() > 0) {
-            final long thisRead = getFrom.pollFirst();
-            ILogData ld = read(thisRead);
-            if (getFrom == context.readQueue) {
-                addToResolvedQueue(context, thisRead, ld);
+        if (useBackpointers) {
+            if (getFrom.size() > 0) {
+                final long thisRead = getFrom.pollFirst();
+                ILogData ld = read(thisRead);
+                if (getFrom == context.readQueue) {
+                    addToResolvedQueue(context, thisRead, ld);
+                }
+                return ld;
             }
-            return ld;
+        } else {
+            return getFromAddressMap(context, getFrom);
         }
 
         // None of the potential reads ended up being part of this
         // stream, so we return null.
         return null;
     }
 
+    private ILogData getFromAddressMap(QueuedStreamContext context, NavigableSet<Long> getFrom) {
```

Comment:
Why this method is so overcomplicated? So many enclose `if` `else` conditions. It's very complicated and error-prone.
Can you please refactor this and simplify the logic.